runtime/parachains/src/inclusion.rs

```diff
@@ -739,6 +757,23 @@ impl<T: Trait> CandidateCheckContext<T> {
 			),
 			Error::<T>::InvalidUpwardMessages,
 		);
+		ensure!(
+			<router::Module<T>>::check_hrmp_watermark(
+				para_id,
+				self.relay_parent_number,
+				// TODO: Hmm, we should settle on a single represenation of T::BlockNumber
+				T::BlockNumber::from(hrmp_watermark),
```

Comment:
Makes sense although a bit annoying, I wish we used a concrete type. I am going to introduce an error type in #1898 and it would require parameterizing that enum with `BlockNumber` and that seems excessive to me.